systemd-boot/default-settings

```diff
@@ -0,0 +1,40 @@
+#! /usr/bin/sh
+
+# Settings from /etc/sysconfig/filename are available as environment vars
+# with the name 'SYS__FILENAME__KEY' (filename converted to upper case).
+#
+# Not all files are parsed, current list is:
+#   bootloader, language
+#
+
+# include common functions
+. "$PBL_INCLUDE/library"
+
+target=$(uname -m)
+
+case "$target" in
+  x86_64|i?86)
+    image=vmlinuz ;;
+  ppc*)
+    image=vmlinux ;;
+  s390*)
+    image=image ;;
+  armv*)
+    image=zImage ;;
+  aarch64|riscv64)
+    image=Image ;;
+  *)
+    echo "Architecture $target not supported."
+esac
+
+lib_get_cmdline_bls
+
+kernel=$(readlink -f "/boot/$image")
```

Comment:
This expect links in /boot but with sdbootutil they are not present.  To find the default and active entries, bootctl should be used (or read the EFI variables, both in sd-boot and grub2-bls)

Reply:
sdbootutil can give you also the same information with --only-default, but bootctl seems better approach